gobblin-runtime/src/main/java/org/apache/gobblin/runtime/StreamModelTaskRunner.java

```diff
@@ -86,7 +87,10 @@ protected void run() throws Exception {
 
     // The cancel is not propagated to the extractor's record generator when it has been turned into a hot Flowable
     // by publish, so set the shutdownRequested flag on cancel to stop the extractor
-    Flowable streamWithShutdownOnCancel = connectableStream.doOnCancel(() -> this.shutdownRequested.set(true));
+    Flowable streamWithShutdownOnCancel = connectableStream.doOnCancel(() -> {
+      this.shutdownRequested.set(true);
+      this.extractor.shutdown();
```

Comment:
Shouldn't the extractor shutdown be called as a result of setting the shutdownRequestedFlag?

Reply:
only when we request a new record in publisher,  the extractor will shutdown. But in some edge case, i.e. consumer is really slow and we have meet the buffer and not requests record anymore. So I add this to force/make sure executor.shutdown() is called.

Reply:
I don't quite remember how was the failure propagated when the slow write happens, but feels like the right fix here is to add finally block in the place where `extractor.shutdown()` is called in `Task` (e.g. line 514) after catching failure, instead of forcing extractor closure here.